assets/js/ajax-filters.js

```diff
@@ -1,288 +1,458 @@
 /* global job_manager_ajax_filters */
-jQuery( document ).ready( function ( $ ) {
+jQuery( document ).ready( function( $ ) {
+	function job_manager_supports_html5_session_storage() {
+		return window.sessionStorage && typeof window.sessionStorage.setItem === 'function';
+	}
 
-	var xhr = [];
+	var session_storage_prefix = 'job_listing_';
 
-	$( '.job_listings' ).on( 'update_results', function ( event, page, append, loading_previous ) {
-		var data         = '';
-		var target       = $( this );
-		var form         = target.find( '.job_filters' );
-		var showing      = target.find( '.showing_jobs' );
-		var results      = target.find( '.job_listings' );
-		var per_page     = target.data( 'per_page' );
-		var orderby      = target.data( 'orderby' );
-		var order        = target.data( 'order' );
-		var featured     = target.data( 'featured' );
-		var filled       = target.data( 'filled' );
-		var job_types    = target.data( 'job_types' );
-		var post_status  = target.data( 'post_status' );
-		var index        = $( 'div.job_listings' ).index(this);
-		var categories, keywords, location;
-
-		if ( index < 0 ) {
-			return;
+	function job_manager_store_state( $target, state ) {
```

Comment:
Could we add some comments for these functions? Even brief ones would be fine. It's not immediately clear (to me, anyway) what the purpose of each function is.

Reply:
Done in 38b06765d730da12f993c82916a9ba82c9d3eb3f